schemas/manifest-schema.json

```diff
@@ -0,0 +1,174 @@
+        "version": {
+          "type": "string",
+          "pattern": "^\\w[a-zA-Z@#0-9.]*$"
+        },
+        "url": {
+          "type": "string"
+        },
+        "doc": {
+          "type": "string"
+        }
+      },
+      "required": ["name", "version"],
+      "additionalProperties": false
+    },
+    "imports": {
+      "type": "object",
+      "properties": {
+      },
+      "propertyNames": {
+        "pattern": "^[A-Za-z_][A-Za-z0-9_]*$"
+      },
+      "additionalProperties": {
+        "type": "string",
+        "pattern": "^\\w[a-zA-Z@#0-9.]*$"
+      },
+      "minProperties": 1
+    },
+    "protobuf": {
+      "type": "object",
+      "properties": {
+        "files": {
+          "type": "array",
+          "items": {
+              "type": "string",
+              "pattern": "^\\w[a-zA-Z@#0-9.]*$"
+          },
+          "minItems": 1,
+          "unevaluatedItems": false
+        },
+        "importPaths": {
+          "type": "array",
+          "prefixFiles": [
+            {
+              "type":  "string",
+              "pattern": "^\\w[a-zA-Z@#0-9.]*$"
+            }
+          ],
+          "minItems": 1
+        }
+      },
+      "required": ["files", "importPaths"],
+      "unevaluatedProperties": false
+    },
+    "binaries" : {
+      "type": "object",
+      "properties": {
+        "default" : {
+          "type": "object",
+          "properties": {
+            "type": {
+              "enum": ["wasm/rust-v1"]
+            },
+            "file": {
+              "type": "string"
+            },
+            "protoPackageMapping" : {
+              "type": "object",
+              "properties": {
+              },
+              "minProperties": 1
+            }
+          },
+          "required": ["type", "file"],
+          "additionalProperties": false
+        }
+      },
+      "required": ["default"],
+      "unevaluatedProperties": false
+    },
+    "modules": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "name": {
+            "type": "string"
+          },
+          "initialBlock": {
+            "type": "number"
+          },
+          "kind": {
+            "enum": [
+              "map",
+              "store"
+            ]
+          },
+          "binary": {
+            "type": "string"
+          },
+          "inputs": {
+            "type": "array",
+            "items": {
+              "type": "object",
+              "minProperties": 1,
+              "maxProperties": 2,
+              "properties": {
+                "store": {
+                  "type": "string"
+                },
+                "source": {
+                  "type": "string"
+                },
+                "map": {
+                  "type": "string"
+                },
+                "mode": {
+                  "type": "string"
+                }
+              },
+              "additionalProperties": false
+            },
```

Comment:
I think it is better modelled with `"items": { "oneOf": [ { "type": "object", "properties": { ... }, ... }, ... ]` because not every combination of properties is valid `mode` for example can only be applied via `store` property.